sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -60,10 +60,16 @@ object StringUtils extends Logging {
         case c if c == escapeChar => fail("it is not allowed to end with the escape character")
         case '_' => out ++= "."
         case '%' => out ++= ".*"
+        case '*' if filterPattern => out ++= ".*"
+        case '.' if filterPattern => out ++= "."
```

Comment:
The title of this PR seems to support for SQL wildcard characters (% and _) , why we added '*' and '.' ?